src/third/redux-persist/getStoredState.js

```diff
@@ -18,27 +18,30 @@ export default function getStoredState (config, onComplete) {
   let restoredState = {}
   let completionCount = 0
 
-  storage.getAllKeys((err, allKeys) => {
-    if (err) {
+  storage.getAllKeys()
+    .catch(err => {
       if (process.env.NODE_ENV !== 'production') console.warn('redux-persist/getStoredState: Error in storage.getAllKeys')
       complete(err)
-      return
-    }
-
-    let persistKeys = allKeys.filter((key) => key.indexOf(keyPrefix) === 0).map((key) => key.slice(keyPrefix.length))
-    let keysToRestore = persistKeys.filter(passWhitelistBlacklist)
+    })
+    .then(allKeys => {
+      let persistKeys = allKeys.filter((key) => key.indexOf(keyPrefix) === 0).map((key) => key.slice(keyPrefix.length))
+      let keysToRestore = persistKeys.filter(passWhitelistBlacklist)
 
-    let restoreCount = keysToRestore.length
-    if (restoreCount === 0) complete(null, restoredState)
-    keysToRestore.forEach((key) => {
-      storage.getItem(createStorageKey(key), (err, serialized) => {
-        if (err && process.env.NODE_ENV !== 'production') console.warn('redux-persist/getStoredState: Error restoring data for key:', key, err)
-        else restoredState[key] = rehydrate(key, serialized)
-        completionCount += 1
-        if (completionCount === restoreCount) complete(null, restoredState)
+      let restoreCount = keysToRestore.length
+      if (restoreCount === 0) complete(null, restoredState)
+      keysToRestore.forEach(async (key) => {
+        try {
+          const serialized = await storage.getItem(createStorageKey(key))
+          // `rehydrate` won't throw an error; see implementation
```

Comment:
A comment like this sounds ominously like famous last words to me. :grin: 

... Looking at the implementation, with that try/catch right around the outside, it does look pretty solid. But still, as a non-local condition it makes me nervous, because that could change.

Is this a point that's needed for understanding the code as it ends up? I think it's really only part of your reasoning for why the change this commit makes is NFC. In that case it'd be better not to leave it lying around in the code anyway, and keep it just in the commit message.

Reply:
My next revision will should make it irrelevant to the NFC-ness whether or not `rehydrate` might throw an error.